src/steps/resource-manager/key-vault/index.test.ts

```diff
@@ -211,3 +220,159 @@ describe('rm-keyvault-principal-relationships', () => {
     ]);
   }, 10_000);
 });
+
+describe('step = key vault keys', () => {
+  beforeAll(async () => {
+    instanceConfig = {
+      clientId: process.env.CLIENT_ID || 'clientId',
+      clientSecret: process.env.CLIENT_SECRET || 'clientSecret',
+      directoryId: '4a17becb-fb42-4633-b5c8-5ab66f28d195',
+      subscriptionId: '87f62f44-9dad-4284-a08f-f2fb3d8b528a',
+      developerId: 'keionned',
+    };
+
+    context = createMockAzureStepExecutionContext({
+      instanceConfig,
+      entities: [
+        {
+          _key: `/subscriptions/${instanceConfig.subscriptionId}`,
+          _class: ['Account'],
+          _type: 'azure_subscription',
+          id: `/subscriptions/${instanceConfig.subscriptionId}`,
+        },
+        {
+          _key: `/subscriptions/${instanceConfig.subscriptionId}/resourceGroups/j1dev`,
+          _type: 'azure_resource_group',
+          _class: ['Group'],
+          name: 'j1dev',
+          location: 'eastus',
+          id: `/subscriptions/${instanceConfig.subscriptionId}/resourceGroups/j1dev`,
+        },
+      ],
+      setData: {
+        [ACCOUNT_ENTITY_TYPE]: {
+          defaultDomain: 'www.fake-domain.com',
+          _type: ACCOUNT_ENTITY_TYPE,
+          _key: 'azure_account_id',
+          id: 'azure_account_id',
+        },
+      },
+    });
+
+    recording = setupAzureRecording({
+      directory: __dirname,
+      name: 'resource-manager-step-key-vaults',
+    });
+
+    await fetchKeyVaults(context);
+    await fetchKeyVaultKeys(context);
+  });
+
+  afterAll(async () => {
+    if (recording) {
+      await recording.stop();
+    }
+  });
+
+  it('should collect an Azure Key Vault Key entity', () => {
+    const { collectedEntities } = context.jobState;
+
+    expect(collectedEntities).toContainEqual(
+      expect.objectContaining({
+        _class: KEY_VAULT_KEY_ENTITY_CLASS,
+        _type: KEY_VAULT_KEY_ENTITY_TYPE,
+      }),
+    );
+  });
+
+  it('should collect an Azure Key Vault contains Azure Key Vault Key relationship', () => {
+    const { collectedRelationships } = context.jobState;
+    expect(collectedRelationships).toContainEqual({
+      _class: 'CONTAINS',
+      _fromEntityKey:
+        '/subscriptions/${instanceConfig.subscriptionId}/resourceGroups/j1dev/providers/Microsoft.KeyVault/vaults/${instanceConfig.developerId}1-j1dev',
```

Comment:
I have kept the tests in the old format for now, as you suggested. They are giving us all sorts of errors atm (400, 401) but I don't think these are _'right'_.  

Look at `/resourceGroups/j1dev/`. I don't think we are using this resource group/config anymore. I don't think this test will pass. It doesn't make a lot of sense. 

You will see a lot of keys & ids being statically typed throughout this file. Same issue with all of them.

Reply:
I think the 4xx errors that we are seeing could possibly be a result of this 'static' configuration.